SplitIt/app/src/main/java/project/cs495/splitit/GroupViewActivity.java

```diff
@@ -0,0 +1,249 @@
+package project.cs495.splitit;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.PopupMenu;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.view.LayoutInflater;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.firebase.ui.database.FirebaseRecyclerAdapter;
+import com.firebase.ui.database.FirebaseRecyclerOptions;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.Query;
+import com.google.firebase.database.ValueEventListener;
+
+import project.cs495.splitit.models.Group;
+import project.cs495.splitit.models.Item;
+import project.cs495.splitit.models.User;
+
+public class GroupViewActivity extends AppCompatActivity implements PopupMenu.OnMenuItemClickListener {
+    private static final String TAG = "GroupViewActivity";
+    public static final String EXTRA_GROUP_ID = "project.cs495.splitit.GROUP_ID";
+    private DatabaseReference mDatabase;
+    private RecyclerView groupRV;
+    private FirebaseRecyclerAdapter adapter;
+    private String groupId;
+    private Group group;
+    private ImageButton fab_plus;
+    private View view;
+    private static int currGroupIndex = 0;
+    private String currGroupId;
+
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_group_view);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar2);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setTitle(R.string.group);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        Intent intent = getIntent();
+        groupId = intent.getStringExtra(EXTRA_GROUP_ID);
+        mDatabase = FirebaseDatabase.getInstance().getReference();
+        mDatabase.child("groups").addListenerForSingleValueEvent(new ValueEventListener() {
```

Comment:
This should use a direct query instead of iterate over children then breaking.  Path should be mDatabase.child("groups").child(groupId).setListener....